src/main/java/io/nats/client/Connection.java

```diff
@@ -562,6 +558,52 @@ enum Status {
      */
     Duration RTT() throws IOException;
 
+    /**
+     * Gets a stream context.
+     * @param streamName the stream for the context
+     * @return a StreamContext instance.
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     */
+    StreamContext streamContext(String streamName) throws IOException, JetStreamApiException;

```

Comment:
I am not sure it's valuable to have stream/consumer methods on `Connection` - they seem to just clutter the interface, or is there any clear benefit?

Reply:
@piotrpio My thought was I don't need a JetStreamContext unless I'm publishing, so it's less steps to get it from the connection, so hanging off connection is my first choice and hanging off JetStream is the second choice. Especially since StreamContext has a lot of the same stuff as JetStreamManagement, it made sense to have it at connection level.
But I was never really settled on it either way, so I guess let's continue to think about this.